### ISSUE TITLE
install rsync compatible with MSYS2 in Windows templates

### ISSUE THREAD

**Issue by ashkulz** [mcandre/packer-templates#169]
The package `rsync` installed via chocolatey in `windows/fix-synced-folders.ps1` uses paths which are inconsistent with the rest of the system. I've found a workaround by using rsync from MSYS2: executing the following in the provisioning step implements it:
```
choco uninstall -y rsync
cd "C:\Program Files\Git"
curl -L https://downloads.sourceforge.net/msys2/rsync-3.1.3-1-x86_64.pkg.tar.xz | tar xJ
```

As Git uses MSYS2, this works out pretty well -- the SSH subsystem uses `/c/whatever` consistently. For 32-bit WIndows, the URL will change slightly -- it will be `i686` instead of `x86_64`. Not sure how to incorporate it for the general case, though -- although ideally there should be a chocolatey package which installs the compatible rsync I'm not too sure it'd be accepted as it has direct dependencies on the Git installation. If you want, I'll try to dig up a PR which changes this locally after some research.

**Comment by mcandre** [comment]
Yeah, tell me more about how MSYS2's rsync behaves differently than Chocolatey's rsync. I'm most concerned with how this would impact the Windows buildbots I'm configuring for [tonixxx](https://github.com/mcandre/tonixxx), so the details are definitely important to me.

I'm also interested in exactly what MSYS2's rsync's dependencies would be. Chocolatey's rsync is clearly dependent on Chocolatey and not much else. Does MSYS2's rsync depend on MSYS2 being installed? Is there another way to inflate tar.xz archives in Windows without requiring tar? Feel free to open a PR and experiment. You can also fork and/or host your own packer templates for Windows, and customize at will!

**Comment by ashkulz** [comment]
Because `git` is already installed and uses MSYS2, I [extract it in that location](https://github.com/wkhtmltopdf/packaging/blob/6644141d78efd698b95c1e734498d9a10fe98fb5/provision/Vagrantfile#L67) and that's all I need to do. There are no behavioral differences than the other MSYS2/git tools, they use `/c/whatever` -- I use the `tar` command over there to extract tar.xz archives.

I'll probably do a PR including documentation in a week or two, haven't really used packer before this so will need to set up everything.
